From 3d5cae7afed2f93f0c3a61526c2f3bf6a7b73842 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Mon, 19 Apr 2010 14:21:55 +0000
Subject: Skip malformed data from the backend side

Sometimes we may get unexpected data from dbus which was leading
to an endless loop.

See bug 614544 for reference.
---
diff --git a/common/gvfsdaemonprotocol.c b/common/gvfsdaemonprotocol.c
index 3e6a3ca..7feefeb 100644
--- a/common/gvfsdaemonprotocol.c
+++ b/common/gvfsdaemonprotocol.c
@@ -519,6 +519,7 @@ _g_dbus_get_file_info (DBusMessageIter *iter,
   g_object_unref (info);
   g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
 		       _("Invalid file info format"));
+  dbus_message_iter_next (iter);
   return NULL;
 }
 
--
cgit v0.8.3.1
